### DISCUSSION TITLE
"Browser integration" inner workings

### DISCUSSION THREAD

**Discussion by stdedos** [keepassxreboot/keepassxc#7401] [Q&A] [answered]
I am using https://github.com/Frederick888/git-credential-keepassxc to access secrets from the command line.

I am calling:
```bash
printf 'url=https://github.com/\nusername=%s\n' "Github Bot" | git-credential-keepassxc get
```

and I am greeted with this dialog:
![image](https://user-images.githubusercontent.com/98734638/153360232-4d86ef93-c877-433c-bc01-103284bb18a0.png)

Then, I respond to it "Deny All".
However, KeePassXC somehow replies with a different entry, whose URL is matching, but the username does not match at all that of the request.

What's up? Why am I getting a match that does not match at all?

**Comment by droidmonkey** [comment, marked as answer]
You have already allowed that entry to go through without confirmation. Delete the browser settings in the custom data (properties tab) of the entry in question.

**Reply by stdedos** [reply]
Isn't there a "stricter" mode, that the entry only gets what it requests, or empty?

What happens to the entries that do not have a URL?

**Reply by droidmonkey** [reply]
If there are no urls defined then it will not be returned. Strict mode is on by default which is why you see the confirmation prompt. 2.7.0 adds a browser report under database reports to list entries findable by the browser service and their urls and special settings.

**Reply by stdedos** [reply]
> Strict mode is on by default

No such 'strict' found in https://keepassxc.org/docs/KeePassXC_UserGuide.html 😕

**Reply by droidmonkey** [reply]
That's because it is the default. There is only an option to disable it. If you want to deny access to entries you previously "always allowed" then you need to go to that entry and remove that setting.

**Reply by stdedos** [reply]
All I want is to give a comma-separated string of options (e.g. `username=*glob-pattern*`), and be able to access the entry that matches the glob-pattern that I have provided in the command line 😅

**Reply by droidmonkey** [reply]
Then you want to use the keepassxc-cli

**Reply by stdedos** [reply]
No, because https://github.com/keepassxreboot/keepassxc/issues/4212 😕

**Reply by droidmonkey** [reply]
Search was implemented on develop branch and will be in 2.7.0. You can use a snapshot build to try it out.

**Reply by stdedos** [reply]
PR/Issue ID?

**Reply by droidmonkey** [reply]
https://github.com/keepassxreboot/keepassxc/pull/6805
